src/pages/ProductList/ProductList.js

```diff
@@ -2,23 +2,48 @@ import React from 'react';
 import ProductPreview from './ProductPreview/ProductPreview';
 import './ProductList.scss';
 
+const userToken =
+  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6MX0.qywu0fsg1ylVPyh359QAGGFq66TM839qyr-W0_EZT-s';
+
 class ProductList extends React.Component {
   state = {
     productList: [],
   };
 
   componentDidUpdate(prevProps) {
     if (this.props.listId !== prevProps.listId) {
-      fetch(`http://localhost:3000/data/productData${this.props.listId}.json`, {
-        method: 'GET',
-      })
-        .then(res => res.json())
-        .then(data => this.setState({ productList: data }));
+      if (localStorage.getItem('token') === null) {
+        fetch(
+          `https://f960-211-106-114-186.ngrok.io/product/menu/${this.props.listId}/navbar`,
+          {
+            method: 'GET',
+          }
+        )
+          .then(res => res.json())
+          .then(data => {
+            this.setState({ productList: data.result });
+          });
+      } else {
+        fetch(
+          `https://f960-211-106-114-186.ngrok.io/product/menu/${this.props.listId}/navbar`,
+          {
+            method: 'GET',
+            headers: {
+              'Content-type': 'application/json',
+              Authorization: localStorage.getItem('token'),
+            },
+          }
+        )
+          .then(res => res.json())
+          .then(data => {
+            this.setState({ productList: data.result });
+          });
+      }
```

Comment:
이 부분은 토큰의 유무에 따라 다른 data를 받아오는 부분인가요?
이 코드만 봤을 때, 이렇게도 써볼 수 있을 것 같습니다.
```suggestion
      const TOKEN = localStorage.getItem('token');
      
      fetch(
          `https://f960-211-106-114-186.ngrok.io/product/menu/${this.props.listId}/navbar`,
          {
            method: 'GET',
            ...(TOKEN && { headers: {
              'Content-type': 'application/json',
              Authorization: TOKEN,
            }}),
          }
        )
          .then(res => res.json())
          .then(data => {
            this.setState({ productList: data.result });
          });
```

Reply:
그리고 `'Content-type': 'application/json'` 을 명시해주신 이유가 있을까요!?